src/query/functions-v2/src/scalars/comparison.rs

```diff
@@ -236,19 +236,37 @@ fn register_like(registry: &mut FunctionRegistry) {
         FunctionProperty::default(),
         |_, _| None,
         vectorize_regexp(|str, pat, map, _| {
+            let mut sub_strings: Vec<&str> = vec![];
             let pattern = if let Some(pattern) = map.get(pat) {
                 pattern
             } else {
                 let pattern_str = simdutf8::basic::from_utf8(pat).map_err(|err| {
```

Comment:
These codes are inside the whole loop, need to have faster path.

Reply:
> These codes are inside the whole loop, need to have faster path.

Got